.github/workflows/tests.yml

```
@@ -0,0 +1,50 @@
+
+      - name: Run ruff
+        run: |
+          ruff check .
+
+      - name: Run pytest
+        run: |
+          pytest --cov=src --cov-report=xml --cov-report=term
+
+      - name: Upload coverage to Codecov
+        uses: codecov/codecov-action@v4
+if: matrix.python-version == '3.11'        with:
```

actionlint (1.7.10)

[error] could not parse as YAML: mapping values are not allowed in this context

(syntax-check)